### DISCUSSION TITLE
Need help using one-click installs - Heroku

### DISCUSSION THREAD

**Discussion by Mbengparis** [directus/directus#5900] [Archive]
I am trying to deploy directus to Heroku using the One-Click Installs but it keeps failing on  Run scripts & scale dynos.  The build is successful but it fails on  Run scripts & scale dynos.

I dont get any error log when it fails for me to try to tracee the problem
For the admin emial and password i used my directus admin login.

Then registerd to aws and used the credentials to fill up the template during one click install. 
Maybe i am be missing something. Do i need to do any other configurations for the deeploy to be succesfull ? 

It is also not really clear in the docs if there are extra setups that have to be done.

Please i need help.

**Comment by rijkvanzanten** [comment]
> [...] but it fails on [...]

How does it fail?

> Then registerd to aws and used the credentials to fill up the template during one click install.

AWS? What does that have to do with the Heroku setup?

I'm a little confused about what you're trying to do or where/how/when it fails 😅

**Reply by Mbengparis** [reply]
@rijkvanzanten  using the one click install we required to input the following fields. Check the pics. 

Where do i get the following values to fill in the required fields?

STORAGE_LOCATIONS
STORAGE_CLOUD_DRIVER
STORAGE_CLOUD_KEY
STORAGE_CLOUD_SECRET
STORAGE_CLOUD_BUCKET
STORAGE_CLOUD_REGION
STORAGE_CLOUD_PUBLIC_URL
STORAGE_CLOUD_ENDPOINT

I believed the above values are to be goten from AWS if I want to use AWS for cloud storage  or am I thinking wrong ?.

Thanks in advance 

![A77B271D-5998-4CFE-BBF5-26DA49594768](https://user-images.githubusercontent.com/49481570/119731605-6fbd6180-be77-11eb-81ff-6da259692ce2.png)
![A5179E59-48E5-4DDB-8F86-5F2C93E6E7B6](https://user-images.githubusercontent.com/49481570/119731620-7350e880-be77-11eb-949c-4d6fe8e9ebf5.png)

**Comment by Mbengparis** [comment]
@rijkvanzanten please check the pics.

I need help deploying directus on heroku

**Reply by rijkvanzanten** [reply]
> Where do i get the following values to fill in the required fields?

You can't store files on a Heroku app instance, so you'll have to plugin an external file storage adapter. You can use AWS S3, DigitalOcean Spaces, Google Cloud Storage, or any other S3 compatible storage provider 👍🏻

**Reply by Mbengparis** [reply]
I did use the AWS S3 - Relational Database Service (RDS) (Aurora) and the configurations have been filled but i get the following error. 

I connected the RDS database  to Sequel Pro and it connected but don't  know why i get the following error that it can not connect to the database when deploying to heroku.

<img width="845" alt="Screenshot 2021-05-27 at 17 05 06" src="https://user-images.githubusercontent.com/49481570/119850725-17856e80-bf0e-11eb-94f3-c09c8a26cdca.png">

**Reply by rijkvanzanten** [reply]
Make sure you have all the permissions configured on AWS, Heroku seems to have a knowledge base article on the matter: https://devcenter.heroku.com/articles/amazon-rds

**Comment by Mbengparis** [comment]
@rijkvanzanten  following the link you sent how can i add the certificate file during one click install?. Because from what they are saying the app has to first been already deployed to heroku then you download the certificate file and push to the heroku app.

Take a look as stated from the link u sent:
_____________________________________________________________
Add the certificate file to your app’s git repository and redeploy to Heroku.

Update the DATABASE_URL config var to include the sslca parameter pointing to the certificate file in your repository:
heroku config:set DATABASE_URL="mysql2://username:password@hostname/dbname?sslca=config/amazon-rds-ca-cert.pem" -a <app_id>
______________________________________________



When deploying using one click when i checked the config vars of the app being deployeed, i see that the DATABASE_URL is set to the following url:

postgres://nvsdzcadctfusc:6da831f3eb6b49926c298e0a467da819b0aa3583e4517ea9c3cbd54113259f4f@ec2-54-246-89-234.eu-west-1.compute.amazonaws.com:5432/d2jgi1fldk5dgl

The above url doesn't even have my credentials. When i try to edit it, i get an error that it can't be overwritten.
The one click automatically creates a database url, the problem is i don't know where it gets the credentials on the url and it can not also  be chnaged.

**Reply by rijkvanzanten** [reply]
> The above url doesn't even have my credentials.

Yes it does

```
postgres://nvsdzcadctfusc:6da831f3eb6b49926c298e0a467da819b0aa3583e4517ea9c3cbd54113259f4f@ec2-54-246-89-234.eu-west-1.compute.amazonaws.com:5432/d2jgi1fldk5dgl
           ^^^^^^^^^^^^^^:^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^@^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^:^^^^/^^^^^^^^^^^^^^
           Username      : Password                                                       @ Host                                            : Port / Database
```

so make sure to swap out those keys on S3, as you've just shared your credentials in a public forum.

---

I have no experience with Heroku, so I can't help you further with the specific configuration between Heroku-AWS

**Reply by Mbengparis** [reply]
@rijkvanzanten  Those are  not my keys. They are diffenrent. My question is where does the keys come from ?. According to the following example i have to set in my username and so on and on the template during one click install that was never asked. So how am i sure its getting the right credentials. Those credentials or keys are completely different from mine.

heroku config:set DATABASE_URL="mysql2://username:password@hostname/dbname?sslca=config/amazon-rds-ca-cert.pem" -a <app_id>

What i noticed also is that heroku automatically installs its postgres database and uses the credentials which is not how the one click install is supposed to work. That is where the credentails or keys is coming from.

But instead i want it to use but the database connection of my AWS RDS. From the one click install it is also not possible to input your database credentials. 

My AWS RDS database is a mysql database not postgres.

How then can i use my own AWS RDS credentials for the install to work during the one click install?

**Reply by rijkvanzanten** [reply]
No idea, like I mentioned above:

> I have no experience with Heroku, so I can't help you further with the specific configuration between Heroku-AWS

**Comment by SapiensEduvision** [comment]
I just deployed Directus on Heroku, but without 1 click install button, I was having trouble with that too.
What worked for my was:

1. Make a local install of directus, using npx create-directus-project
2. I'm using MySQL instance on rds, so on DBHOST prompt just put rds endpoint and then the credentials
3. Test local install is working
4. Init a git repo on the project, ignore files that you dont require
5. Upload to github
6. On Heroku create a new app and follow the instructions 
7. On deployment method connect your github account and select your project

An interesting thing here is that I didn't configure any storage driver, just left that to local, and is working but you can put aws s3 following this structure https://docs.directus.io/reference/environment-variables/#s3-s3

That worked for me

**Reply by rijkvanzanten** [reply]
> An interesting thing here is that I didn't configure any storage driver, just left that to local, and is working

Keep in mind that this means that it's storing files in the Heroku container it's running on. When you have more than 1 instance running, or if Heroku destroys / recreates that container on updates or crashes, you'll run into data persistency issues

**Reply by SapiensEduvision** [reply]
You right, just tried and yes, uploaded files and after restart the dynos, files get corrupted

**Reply by Mbengparis** [reply]
@SapiensEduvision when u deploy directus by uploading first to github, does your heroku URL page look like this ?

Using local with any storage driver works but when i configure it to us the AWS RDS i get CORS error .

i don't know why because my frontend is also deployed to heroku.

<img width="1253" alt="Screenshot 2021-05-29 at 00 49 15" src="https://user-images.githubusercontent.com/49481570/120048775-4be55180-c018-11eb-9c00-09b646182408.png">

**Reply by SapiensEduvision** [reply]
@Mbengparis No, it shows Directus Admin app, Have you set right inbound/outbound rules on RDS security group instance?

**Comment by Mbengparis** [comment]
Thanks so much I will try that
